rdkit/Chem/Pharm3D/EmbedLib.py

```diff
@@ -1259,6 +1364,309 @@ def ComputeChiralVolume(mol, centerIdx, confId=-1):
+    except ValueError as e:
+      return []    # This is a bogus mapping
+    else:
+      if len(embeddings) == 0:
+        return []  # This is a bogus mapping
+
+    coords = [embed.GetConformer(0).GetPositions() for embed in embeddings]
+    for coordTuple in coords:
+      ph4_constraints.append(coordTuple)
+    return ph4_constraints
+
+  def get_conformers_from_ph4_constraint(self, ph4_constraint, mapping, pharmacophore):
+    if self.settings.usePh4Constraints:
+      embedding = self._embed_conformers_constrained(ph4_constraint, mapping)
+    else:
+      embedding = self._embed_conformers_unconstrained()
+    if embedding is None:
+      return None
+    return GetAlignedConformers(embedding, pharmacophore, mapping, self.settings.minimize)
+
+  def _embed_conformers_unconstrained(self):
+    self.ligand.RemoveAllConformers()
+    numConfs = min(GetNumOptimalConformers(self.ligand), self.settings.maxConformers)
+    mol = Chem.Mol(self.ligand)
+
+    if self.settings.useETKDG:
+      # FIX: EmbedMultipleConfs args != EmbedParameters fields
+      rdDistGeom.EmbedMultipleConfs(mol, numConfs, rdDistGeom.ETKDG())
+    else:
+      rdDistGeom.EmbedMultipleConfs(mol, numConfs)
+    if mol.GetNumConformers() > 0:
+      return mol
+
+  def _embed_conformers_constrained(self, ph4_constraint, mapping):
+    self.ligand.RemoveAllConformers()
+    mol = Chem.Mol(self.ligand)
+    numConfs = max(1,
+                   min(GetNumOptimalConformers(self.ligand),
+                       self.settings.maxConformers)
+                   // self.settings.maxMappings
+                   // self.settings.maxEmbeddings) # enforce floor int division
+
+    # Currently we select the first atom from each MolFeature. This can
+    # (and should) be changed.
+    staticAts = [molFeat[0] for molFeat in mapping]
+    coordMap = {i:Geometry.Point3D(*ph4_constraint[i]) for i in range(
+          len(mapping)) if i in staticAts}
+    assert len(coordMap) > 0
+
+    if self.settings.useETKDG:
+      # params = rdDistGeom.ETKDG()
+      # params.coordMap = coordMap  # for now, need to manually specify coordMap
+      rdDistGeom.EmbedMultipleConfs(
+          mol,        # mol
+          numConfs,   # numConfs
+          0,          # maxAttempts
+          -1,         # randomSeed
+          True,       # clearConfs
+          False,      # useRandomCoords
+          2.0,        # boxSizeMult
+          True,       # randNegEig
+          1,          # numZeroFail
+          -1.0,       # pruneRmsThresh
+          coordMap,   # coordMap
+          1e-3,       # forceTol
+          False,      # ignoreSmoothingFailures
+          True,       # enforceChirality
+          1,          # numThreads
+          True,       # useExpTorsionAnglePrefs
+          True,       # useBasicKnowledge
+          False)      # printExpTorsionAngles
+    else:
+      nAts = mol.GetNumAtoms()
+      conf = Chem.Conformer(nAts)
```

Comment:
It's not clear to me why you build a conformer and add it to the molecule before you call `EmbedMultipleConfs()`, which will then immediately remove that conformer.